common/init.sls

```diff
@@ -1,11 +1,38 @@
 {% from tpldir ~ '/map.jinja' import config with context %}
 
 {% if grains['kernel'] != 'Darwin' %}
-python-pkgs:
+# Ubuntu has python2 as default python
+python2:
   pkg.installed:
     - pkgs:
-      - python-pip
+      - python
       - python-dev
+
+python3:
+  pkg.installed:
+    - pkgs:
+      - python3
+
+# Ensure pip is default by purging pip3
+pip:
+  pkg.installed:
+    - pkgs:
+      - python-pip
+    - reload_modules: True
+
+pip3:
+  pkg.purged:
+    - pkgs:
+      - python3-pip
+
+# Virtualenv package creates virtualenv and virtualenv-3.4 executables
```

Comment:
Yes, on Ubuntu the defaults are all 2.7. [Ubuntu is not planning to change this until PEP 394 says otherwise](https://wiki.ubuntu.com/Python/3).

`bin_env`: Right, I'm using this but in a different way. The approach I'm taking is to use `virtualenv.managed` and pass in python3 in `homu/init.sls`; salt automatically will install pip3 in the virtualenv (pip3 is bundled with python starting with python3.4). Then, I use pip.installed and set `bin_env` to the virtualenv location for homu's dependencies, which means it ends up using pip3 under the hood.

Also, for future reference you'll want to use the [state documentation](https://docs.saltstack.com/en/latest/ref/states/all/salt.states.pip_state.html), not the module docs :)